### DISCUSSION TITLE
failed end device 9.2.17 with OT_RADIO_CAPS_TRANSMIT_SEC capability

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#8486] [Q&A] [answered]
Hello,

I'm using Thread Test Harness software Release 2.0.0 and I'm integrating platform with openthread cli example

I'm checking end device test case 9.2.17 from thread harness with device that has OT_RADIO_CAPS_TRANSMIT_SEC capability enabled but it always fails. When OT_RADIO_CAPS_TRANSMIT_SEC capability is disabled the test case passes.

I can see that all packets that needs security are already secured by openthread core as when they passed to radio (mIsSecurityProcessed, mIsHeaderUpdated) flags are set , so is this expected ?

I need to know the reason behind this failure.

Attached 2 sniffer captures for it (1 for ch.11 and the other for ch.12) and the report produced from test case
[OpenThread_End Device_Consolidate Report_R2.0.0_2022_12_06-09_59_23.pdf](https://github.com/openthread/openthread/files/10163989/OpenThread_End.Device_Consolidate.Report_R2.0.0_2022_12_06-09_59_23.pdf)
[sniffer_capture_9_2_17.zip](https://github.com/openthread/openthread/files/10164001/sniffer_capture_9_2_17.zip)

Thanks

**Comment by jwhui** [comment, marked as answer]
If the radio is properly projecting the `OT_RADIO_CAPS_TRANSMIT_SEC` capability, then OpenThread core should not be encrypting the frame and instead relying on the radio driver to encrypt.
